src/frontend/src/optimizer/rule/over_window_split_rule.rs

```diff
@@ -19,22 +19,31 @@ use itertools::Itertools;
 use super::Rule;
 use crate::PlanRef;
 
-pub struct OverWindowSplitByWindowRule;
+pub struct OverWindowSplitRule;
 
-impl OverWindowSplitByWindowRule {
+impl OverWindowSplitRule {
     pub fn create() -> Box<dyn Rule> {
-        Box::new(OverWindowSplitByWindowRule)
+        Box::new(OverWindowSplitRule)
     }
 }
 
-impl Rule for OverWindowSplitByWindowRule {
+impl Rule for OverWindowSplitRule {
     fn apply(&self, plan: PlanRef) -> Option<PlanRef> {
         let over_window = plan.as_logical_over_window()?;
+        let mut rank_func_seq = 0;
         let groups: HashMap<_, _> = over_window
             .window_functions()
             .iter()
             .enumerate()
-            .map(|(idx, window)| ((&window.order_by, &window.partition_by), idx))
+            .map(|(idx, func)| {
+                let func_seq = if func.kind.is_rank() {
+                    rank_func_seq += 1;
+                    rank_func_seq
+                } else {
+                    0
+                };
+                ((&func.order_by, &func.partition_by, func_seq), idx)
+            })
```

Comment:
I guess common window function elimination is already done by `LogicalOverWindowBuilder` in #10956, similar to Agg.

Reply:
If window function elimination is already done, does it means `rank_func_seq` is unnecessary?

Reply:
Nope. `row_number`, `rank` and `dense_rank` are all _rank_ function in our terminology and they behave differently. The logic here is for split all these rank functions with same `PARTITION BY` and `ORDER BY` because our `GroupTopN` can only handle one for each.

Reply:
Got it.